Ndev 3834

romanova-natasha opened the pull request.

Commit:
rename proxy image with "evm-triggered-*" when run from evm repo

rename proxy image with "evm-triggered-*" when run from evm repo

Commit:
change: define evm tag logic

change: define evm tag logic

neonlabstech commented:
<details><summary>Dapps report</summary>

🔗[Cost report](https://github.com/neonlabsorg/neon-proxy.py/actions/runs/16422479847/artifacts/3579789925)

neonlabstech commented:
<details><summary><strong>⚠️ Cost Report Validation Failed ⚠️</strong></summary>

| dapp           | action                                          | metric    |   INCREASE |
|:---------------|:------------------------------------------------|:----------|-----------:|
| aave           | borrow from lending pool                        | acc_count |          1 |
| aave           | borrow from lending pool                        | gas_used  |    2604120 |
| aave           | deposit to lending pool                         | acc_count |         29 |
| aave           | deposit to lending pool                         | trx_count |          2 |
| aave           | deposit to lending pool                         | gas_used  |    3696940 |
| aave           | liquidation                                     | gas_used  |    1216120 |
| aave           | repay                                           | acc_count |          1 |
| aave           | token approve                                   | gas_used  |    1162320 |
| aave           | token mint                                      | gas_used  |    1162320 |
| aave           | withdraw                                        | acc_count |          1 |
| saddle finance | add liquidity                                   | gas_used  |    1151820 |
| saddle finance | add liquidity in 3 tokens                       | gas_used  |    1158120 |
| saddle finance | add liquidity in metapool                       | gas_used  |    1158120 |
| saddle finance | swap usdc -> susd metapool                      | acc_count |          1 |
| uniswap v3     | burn transaction                                | gas_used  |     459360 |
| uniswap v3     | nonfungiblepositionmanager - decrease liquidity | gas_used  |     453060 |
| uniswap v3     | nonfungiblepositionmanager - mint position      | gas_used  |   12290140 |
| uniswap v3     | token approve                                   | gas_used  |    1162320 |

neonlabstech commented:
<details><summary>Solana Requests Statistics</summary>



|                                   |   count |   min_time |   max_time |   average_time |   median_time |
|:----------------------------------|--------:|-----------:|-----------:|---------------:|--------------:|
| getAccountInfo                    |   40383 |      0     |      0.239 |    0.000481118 |         0     |
| getBlocks                         |   26692 |      0     |      0.206 |    0.000306272 |         0     |
| getBlock                          |   13495 |      0     |      0.027 |    0.000546943 |         0     |
| getTransaction                    |   12007 |      0     |      0.229 |    0.00213409  |         0.001 |
| getMultipleAccounts               |    8848 |      0     |      0.242 |    0.00133013  |         0.001 |
| sendTransaction                   |    5919 |      0     |      0.238 |    0.00512502  |         0.002 |
| getLatestBlockhash                |    5383 |      0     |      0.034 |    0.000331414 |         0     |
| getBalance                        |    1912 |      0     |      0.009 |    0.000231695 |         0     |
| getSignatureStatuses              |     474 |      0     |      0.031 |    0.000468354 |         0     |
| isBlockhashValid                  |     214 |      0     |      0.001 |    0.000214953 |         0     |
| getVersion                        |     180 |      0     |      0.011 |    0.000261111 |         0     |
| getSignaturesForAddress           |     104 |      0     |      0.002 |    0.000951923 |         0.001 |
| requestAirdrop                    |      42 |      0     |      0.001 |    0.00047619  |         0     |
| simulateTransaction               |      42 |      0     |      0.002 |    0.0005      |         0     |
| getBlockHeight                    |      26 |      0     |      0.01  |    0.000653846 |         0     |
| getSlot                           |      25 |      0     |      0.001 |    0.00052     |         0.001 |
| getTokenAccountBalance            |       3 |      0     |      0.001 |    0.000666667 |         0.001 |
| getFirstAvailableBlock            |       2 |      0     |      0     |    0           |         0     |
| getHealth                         |       2 |      0     |      0     |    0           |         0     |
| getMinimumBalanceForRentExemption |       1 |      0.001 |      0.001 |    0.001       |         0.001 |
| getProgramAccounts                |       1 |      0.038 |      0.038 |    0.038       |         0.038 |
| getClusterNodes                   |       1 |      0.001 |      0.001 |    0.001       |         0.001 |

</details>

neonlabstech commented:
<details><summary><strong>⚠️ Cost Report Validation Failed ⚠️</strong></summary>

| dapp           | action                                          | metric    |   INCREASE |
|:---------------|:------------------------------------------------|:----------|-----------:|
| aave           | borrow from lending pool                        | gas_used  |    2612520 |
| aave           | deposit to lending pool                         | acc_count |         28 |
| aave           | deposit to lending pool                         | trx_count |          2 |
| aave           | deposit to lending pool                         | gas_used  |    3701140 |
| aave           | liquidation                                     | acc_count |          1 |
| aave           | liquidation                                     | gas_used  |    1209820 |
| aave           | token approve                                   | gas_used  |    1160220 |
| aave           | token mint                                      | gas_used  |    1162320 |
| saddle finance | add liquidity                                   | gas_used  |    1156020 |
| saddle finance | add liquidity in 3 tokens                       | gas_used  |    1162320 |
| saddle finance | add liquidity in metapool                       | gas_used  |    1149720 |
| uniswap v3     | burn transaction                                | gas_used  |     459360 |
| uniswap v3     | nonfungiblepositionmanager - decrease liquidity | gas_used  |     453060 |
| uniswap v3     | nonfungiblepositionmanager - mint position      | gas_used  |   12296440 |
| uniswap v3     | token approve                                   | gas_used  |    1162320 |

neonlabstech commented:
<details><summary>Solana Requests Statistics</summary>



|                                   |   count |   min_time |   max_time |   average_time |   median_time |
|:----------------------------------|--------:|-----------:|-----------:|---------------:|--------------:|
| getAccountInfo                    |   40117 |      0     |      0.252 |    0.000499963 |         0     |
| getBlocks                         |   27160 |      0     |      0.213 |    0.000313918 |         0     |
| getBlock                          |   13742 |      0     |      0.074 |    0.000550648 |         0.001 |
| getTransaction                    |   11916 |      0     |      0.071 |    0.00222608  |         0.001 |
| getMultipleAccounts               |    8796 |      0     |      0.255 |    0.00136312  |         0.001 |
| sendTransaction                   |    5871 |      0     |      0.252 |    0.00554727  |         0.002 |
| getLatestBlockhash                |    5368 |      0     |      0.087 |    0.000446162 |         0     |
| getBalance                        |    1903 |      0     |      0.046 |    0.000323699 |         0     |
| getSignatureStatuses              |     479 |      0     |      0.006 |    0.000317328 |         0     |
| isBlockhashValid                  |     217 |      0     |      0.001 |    0.00021659  |         0     |
| getVersion                        |     186 |      0     |      0.003 |    0.00027957  |         0     |
| getSignaturesForAddress           |     100 |      0     |      0.002 |    0.00096     |         0.001 |
| simulateTransaction               |      42 |      0     |      0.001 |    0.000238095 |         0     |
| requestAirdrop                    |      41 |      0     |      0.001 |    0.000487805 |         0     |
| getBlockHeight                    |      26 |      0     |      0.001 |    0.000115385 |         0     |
| getSlot                           |      25 |      0     |      0.001 |    0.00044     |         0     |
| getTokenAccountBalance            |       4 |      0     |      0.001 |    0.00025     |         0     |
| getFirstAvailableBlock            |       2 |      0     |      0     |    0           |         0     |
| getHealth                         |       2 |      0     |      0     |    0           |         0     |
| getMinimumBalanceForRentExemption |       1 |      0     |      0     |    0           |         0     |
| getProgramAccounts                |       1 |      0.031 |      0.031 |    0.031       |         0.031 |
| getClusterNodes                   |       1 |      0     |      0     |    0           |         0     |

</details>

Commit:
name proxy image with prefix "evm-traggered-*" if evm_loader image with proxy branch name exists

name proxy image with prefix "evm-traggered-*" if evm_loader image with proxy branch name exists

neonlabstech commented:
<details><summary><strong>⚠️ Cost Report Validation Failed ⚠️</strong></summary>

| dapp           | action                                          | metric    |   INCREASE |
|:---------------|:------------------------------------------------|:----------|-----------:|
| aave           | borrow from lending pool                        | gas_used  |    2604120 |
| aave           | deposit to lending pool                         | acc_count |         28 |
| aave           | deposit to lending pool                         | trx_count |          2 |
| aave           | deposit to lending pool                         | gas_used  |    3699040 |
| aave           | liquidation                                     | gas_used  |    1207720 |
| aave           | token approve                                   | gas_used  |    1162320 |
| aave           | token mint                                      | gas_used  |    1162320 |
| saddle finance | add liquidity                                   | gas_used  |    1158120 |
| saddle finance | add liquidity in 3 tokens                       | gas_used  |    1160220 |
| saddle finance | add liquidity in metapool                       | gas_used  |    1158120 |
| uniswap v3     | burn transaction                                | gas_used  |     461460 |
| uniswap v3     | nonfungiblepositionmanager - decrease liquidity | gas_used  |     453060 |
| uniswap v3     | nonfungiblepositionmanager - mint position      | gas_used  |   12296440 |
| uniswap v3     | token approve                                   | gas_used  |    1162320 |

Commit:
use image prefix only for triggered pipelines

use image prefix only for triggered pipelines

Commit:
add comment to new logic

add comment to new logic

Commit:
rm unused logic

rm unused logic

neonlabstech commented:
<details><summary><strong>⚠️ Cost Report Validation Failed ⚠️</strong></summary>

| dapp           | action                                          | metric    |   INCREASE |
|:---------------|:------------------------------------------------|:----------|-----------:|
| aave           | borrow from lending pool                        | gas_used  |    2599920 |
| aave           | deposit to lending pool                         | acc_count |         28 |
| aave           | deposit to lending pool                         | trx_count |          2 |
| aave           | deposit to lending pool                         | gas_used  |    3694840 |
| aave           | liquidation                                     | acc_count |          1 |
| aave           | liquidation                                     | gas_used  |    1199320 |
| aave           | token approve                                   | gas_used  |    1164420 |
| aave           | token mint                                      | gas_used  |    1164420 |
| saddle finance | add liquidity                                   | gas_used  |    1151820 |
| saddle finance | add liquidity in 3 tokens                       | gas_used  |    1158120 |
| saddle finance | add liquidity in metapool                       | gas_used  |    1149720 |
| uniswap v3     | burn transaction                                | gas_used  |     459360 |
| uniswap v3     | nonfungiblepositionmanager - decrease liquidity | gas_used  |     453060 |
| uniswap v3     | nonfungiblepositionmanager - mint position      | gas_used  |   12294340 |
| uniswap v3     | token approve                                   | gas_used  |    1160220 |

Commit:
rm useless logic

rm useless logic

neonlabstech commented:
<details><summary>Dapps report</summary>

🔗[Cost report](https://github.com/neonlabsorg/neon-proxy.py/actions/runs/16448086058/artifacts/3588786223)

neonlabstech commented:
<details><summary><strong>⚠️ Cost Report Validation Failed ⚠️</strong></summary>

| dapp           | action                                          | metric    |   INCREASE |
|:---------------|:------------------------------------------------|:----------|-----------:|
| aave           | borrow from lending pool                        | gas_used  |    2606220 |
| aave           | deposit to lending pool                         | acc_count |         28 |
| aave           | deposit to lending pool                         | trx_count |          2 |
| aave           | deposit to lending pool                         | gas_used  |    3699040 |
| aave           | liquidation                                     | gas_used  |    1205620 |
| aave           | token approve                                   | gas_used  |    1160220 |
| aave           | token mint                                      | gas_used  |    1162320 |
| saddle finance | add liquidity                                   | gas_used  |    1153920 |
| saddle finance | add liquidity in 3 tokens                       | gas_used  |    1158120 |
| saddle finance | add liquidity in metapool                       | gas_used  |    1149720 |
| uniswap v3     | burn transaction                                | gas_used  |     459360 |
| uniswap v3     | nonfungiblepositionmanager - decrease liquidity | gas_used  |     453060 |
| uniswap v3     | nonfungiblepositionmanager - mint position      | gas_used  |   12296440 |
| uniswap v3     | token approve                                   | gas_used  |    1160220 |

neonlabstech commented:
<details><summary>Solana Requests Statistics</summary>



|                                   |   count |   min_time |   max_time |   average_time |   median_time |
|:----------------------------------|--------:|-----------:|-----------:|---------------:|--------------:|
| getAccountInfo                    |   40262 |      0     |      0.242 |    0.000521087 |        0      |
| getBlocks                         |   27671 |      0     |      0.206 |    0.000302302 |        0      |
| getBlock                          |   13980 |      0     |      0.063 |    0.000524034 |        0      |
| getTransaction                    |   12057 |      0     |      0.068 |    0.00190819  |        0.001  |
| getMultipleAccounts               |    8847 |      0     |      0.246 |    0.00143246  |        0.001  |
| sendTransaction                   |    5940 |      0     |      0.242 |    0.00534226  |        0.002  |
| getLatestBlockhash                |    5391 |      0     |      0.048 |    0.000398998 |        0      |
| getBalance                        |    1904 |      0     |      0.062 |    0.000296744 |        0      |
| getSignatureStatuses              |     472 |      0     |      0.016 |    0.000400424 |        0      |
| isBlockhashValid                  |     213 |      0     |      0.001 |    0.000201878 |        0      |
| getVersion                        |     179 |      0     |      0.059 |    0.000608939 |        0      |
| getSignaturesForAddress           |     102 |      0     |      0.003 |    0.00112745  |        0.001  |
| requestAirdrop                    |      42 |      0     |      0.001 |    0.000571429 |        0.001  |
| simulateTransaction               |      42 |      0     |      0.001 |    0.000333333 |        0      |
| getSlot                           |      25 |      0     |      0.001 |    0.00044     |        0      |
| getBlockHeight                    |      24 |      0     |      0.001 |    0.000166667 |        0      |
| getTokenAccountBalance            |       3 |      0     |      0     |    0           |        0      |
| getFirstAvailableBlock            |       2 |      0     |      0.001 |    0.0005      |        0.0005 |
| getHealth                         |       2 |      0     |      0     |    0           |        0      |
| getMinimumBalanceForRentExemption |       1 |      0     |      0     |    0           |        0      |
| getProgramAccounts                |       1 |      0.056 |      0.056 |    0.056       |        0.056  |
| getClusterNodes                   |       1 |      0     |      0     |    0           |        0      |

</details>
